### PR TITLE
Made ec2.py dynamic inventory module more modular and importable

### PR DESCRIPTION
**Pull request by adamhadani**
Trying to build off the work in ec2.py, one is tempted to try and have a new script import the stock one and then derive from Ec2Inventory. 
I've actually been using this but had to make a couple slight modifications to the ec2.py script to support this.

This PR is a clone of the original one I created a while back ( #8712 ) that was created after the various ansible modules repositories were branched out.

### CODE REVIEW

File: contrib/inventory/ec2.py [hunk 1]
```diff
@@ -404,7 +406,7 @@ def get_instances_by_region(self, region):
             if e.error_code == 'AuthFailure':
                 error = self.get_auth_error_message()
             else:
-                backend = 'Eucalyptus' if self.eucalyptus else 'AWS'
```
Comment (adamhadani) [inline contrib/inventory/ec2.py:407]: trailing space

File: contrib/inventory/ec2.py [hunk 2]
```diff
@@ -617,7 +619,7 @@ def add_instance(self, instance, region):
                     if self.nested_groups:
                         self.push_group(self.inventory, 'security_groups', key)
             except AttributeError:
-                self.fail_with_error('\n'.join(['Package boto seems a bit older.',
```
Comment (adamhadani) [inline contrib/inventory/ec2.py:620]: trailing space

File: contrib/inventory/ec2.py [hunk 3]
```diff
@@ -714,7 +716,7 @@ def add_rds_instance(self, instance, region):
                         self.push_group(self.inventory, 'security_groups', key)
 
             except AttributeError:
-                self.fail_with_error('\n'.join(['Package boto seems a bit older.',
```
Comment (adamhadani) [inline contrib/inventory/ec2.py:717]: trailing space

**Comment by amenonsen** [comment]
Apart from the extraneous whitespace changes, this looks good to me. It's possible that in future, we may want to make inventory scripts more extensible anyway. I have no idea what that might look like (more plugins?), but this seems as good a place to start as any.

**Comment by ryansb** [comment]
An equivalent change to this has been merged, #21637

**Pull request closed without merging**